packages/web5/src/dwn-api.ts

```diff
@@ -0,0 +1,169 @@
+import type { Web5Agent } from '@tbd54566975/web5-agent';
+import type {
+  MessageReply,
+  RecordsReadOptions,
+  RecordsQueryOptions,
+  RecordsWriteMessage,
+  RecordsWriteOptions,
+  RecordsDeleteOptions,
+  ProtocolsQueryOptions,
+  ProtocolsConfigureOptions,
+} from '@tbd54566975/dwn-sdk-js';
+
+import { DwnInterfaceName, DwnMethodName, DataStream } from '@tbd54566975/dwn-sdk-js';
+
+import { Record } from './record.js';
+import { dataToBytes } from './utils.js';
+
+export type RecordsWriteRequest = {
+  author: string;
+  data: unknown;
+  message?: Omit<RecordsWriteOptions, 'authorizationSignatureInput' | 'data'>;
+}
+
+export type RecordsWriteResponse = MessageReply & {
+  record?: Record
+};
+
+export type RecordsQueryRequest = {
+  author: string;
+  message: Omit<RecordsQueryOptions, 'authorizationSignatureInput'>;
+}
+
+export type RecordsQueryResponse = MessageReply & {
+  records: Record[]
+};
+
+export type RecordsDeleteRequest = {
+  author: string;
+  message: Omit<RecordsDeleteOptions, 'authorizationSignatureInput'>;
+}
+
+export type RecordsDeleteResponse = MessageReply;
+
+export type RecordsReadRequest = {
+  author: string;
+  message: Omit<RecordsReadOptions, 'authorizationSignatureInput'>;
+}
+
+export type RecordsReadResponse = MessageReply;
+
+export type ProtocolsConfigureRequest = {
+  author: string;
+  message: Omit<ProtocolsConfigureOptions, 'authorizationSignatureInput'>;
+}
+
+export type ProtocolsQueryRequest = {
+  author: string;
+  message: Omit<ProtocolsQueryOptions, 'authorizationSignatureInput'>
+}
+
+export const DwnApi = (web5Agent: Web5Agent) => ({
+  protocols: {
+    async configure(target: string, request: ProtocolsConfigureRequest) {
```

Comment:
I see `target` here and I see `tenant` in the `dwn-sdk-js` repo, but is there a distinction with a `did`? As in, a DID is a string (array of bytes, really) of the format defined in [the spec](https://www.w3.org/TR/did-core/), so is `target` and `tenant` some type of superset, or additional functionality of a `did`?

Reply:
The intent behind calling it `target` is in the sense that the methods are all DID-relative with DWN Messaging.  While the DID is represented as a string that can be decoded to bytes, it also is cable of being resolved (see [DID Resolution](https://www.w3.org/TR/did-core/#did-resolution)) to a DID document and the DID document is capable of acting as a routing table through the population of [Service](https://www.w3.org/TR/did-core/#services) entries.

You've likely already read these but for anyone else reading the comments:  More info on how [service endpoints](https://identity.foundation/decentralized-web-node/spec/#service-endpoints) and DID-relative [addressing](https://identity.foundation/decentralized-web-node/spec/#addressing).

 The `target` is the intended destination for which a message is to be routed.  When processed, the target DID is resolved to a specific service endpoint which expresses how the sender of the message is able to communicate with the target entity (person, organization, mediator, etc.).

While it is expected that the target DID specified in a DWN message will be a tenant on a DWeb Node, technically it doesn't HAVE to be a multi-tenancy capable DWN implemented as in `dwn-sdk-js`.  You could even imagine a "bridge" node that exists to relay messages.  On one side its speaking DWeb Messages with DWN nodes and on the other it could be any other protocol (e.g., traditional REST API endpoints, etc.).